lib/m2ts/caption-stream.js

```diff
@@ -626,22 +677,62 @@ Cea708Stream.prototype.initService = function(serviceNum, i) {
  * @param  {Service} service  The service object to be affected
  * @return {Integer}          New index after parsing
  */
-Cea708Stream.prototype.handleText = function(i, service, isExtended) {
+Cea708Stream.prototype.handleText = function(i, service, options) {
+  var isExtended = options && options.isExtended;
+  var isMultiByte = options && options.isMultiByte;
   var packetData = this.current708Packet.data;
-  var b = packetData[i];
   var extended = isExtended ? 0x1000 : 0x0000;
-  var char = get708CharFromCode(extended | b);
+  var currentByte = packetData[i];
+  var nextByte = packetData[i + 1];
   var win = service.currentWindow;
+  var char;
+  var charCodeArray;
+
+  // Use the TextDecoder if one was created for this service
+  if (service.textDecoder_ && !isExtended) {
+    if (isMultiByte) {
+      charCodeArray = [currentByte, nextByte];
```

Comment:
We should subarray here instead to prevent the creation of a new Uint8Array below. 
```
// for multi-byte (can they be more than two characters?)
charCodeArray = packetData.subarray(i, i + 1)

// for single
charCodeArray = packetData.subarray(i)
```

Reply:
Good call

Reply:
@brandonocasey Actually, `packetData` isn't a typed array so we do need to create a Uint8Array here. And to address your other question:
> When a decoder encounters the P16 code, it uses the succeeding **two bytes** to
address characters in a 16-bit code set.